stacker/context.py

```diff
@@ -66,8 +66,26 @@ def namespace_delimiter(self):
 
     @property
     def bucket_name(self):
+        if not self.upload_templates_to_s3:
+            return None
+
         return self.config.stacker_bucket \
-                or "stacker-%s" % (self.get_fqn(),)
+            or "stacker-%s" % (self.get_fqn(),)
+
+    @property
+    def upload_templates_to_s3(self):
+        # Don't upload stack templates to S3 if `stacker_bucket` is explicitly
+        # set to an empty string.
+        if self.config.stacker_bucket == '':
```

Comment:
Can you throw some debug statements into this logic, just so it's easy to debug?